### PR TITLE
fix: fix breadcrumb component bug

### PR DESCRIPTION
**Pull request by ArthurTriis1**
Signed-off-by: Arthur Andrade <arthurfelandrade@gmail.com>

## What's the purpose of this pull request?
This PR fix a Breadcrumb bug. This bug occurs when a conditional rendering is called inside the Breadcrumb, even if the rendering is false the component renders the divider, like in the picture.

Screen Shot 2022-03-24 at 15 45 09

Screen Shot 2022-03-24 at 15 45 55

To fix this bug, the function` React.Children.toArray` was added, it converts from `ReactChildren` to `Array` and filters the `null` childrens

### CODE REVIEW

**Review by eduardoformiga — APPROVED** [review]

**Review by filipewl — APPROVED** [review]
LGTM!

Today I learned about [the difference](https://stackoverflow.com/a/66042465/7970105) between `Children.map` and `Children.toArray.map` (skips null or undefined from conditional rendering).

File: packages/ui/src/molecules/Breadcrumb/Breadcrumb.tsx [hunk 1]
```diff
@@ -88,20 +86,22 @@ const Breadcrumb = forwardRef<HTMLDivElement, BreadcrumbProps>(
         {...otherProps}
       >
         <List data-breadcrumb-list variant="ordered">
-          {React.Children.map(children, (child, index) => {
-            const isLastItem = index === lastIndex - 1
+          {React.Children.toArray(children).map(
```
Comment (igorbrasileiro) [inline packages/ui/src/molecules/Breadcrumb/Breadcrumb.tsx:89]: WDYT about adding some tests for it?

Comment (ArthurTriis1) [reply packages/ui/src/molecules/Breadcrumb/Breadcrumb.tsx:89]: OK! I will!

File: packages/ui/src/molecules/Breadcrumb/Breadcrumb.test.tsx [hunk 2]
```diff
@@ -95,6 +95,20 @@ describe('Breadcrumb', () => {
     expect(dividers).toHaveLength(2)
   })
 
+  it('Should not render null values', () => {
```
Comment (igorbrasileiro) [inline packages/ui/src/molecules/Breadcrumb/Breadcrumb.test.tsx:98]: WDYT about adding tests to check the last divider? Since you changed the value?

**Review by igorbrasileiro — APPROVED** [review]

File: packages/ui/src/molecules/Breadcrumb/Breadcrumb.test.tsx (hunk 2 above)
Comment (ArthurTriis1) [reply packages/ui/src/molecules/Breadcrumb/Breadcrumb.test.tsx:98]: I didn't change the behaviour, just the implementation, so tests that exists already test this

**Pull request merged**